ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ConnectServerErrors.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.server.execution;
+
+import static io.netty.handler.codec.http.HttpResponseStatus.FORBIDDEN;
+import static io.netty.handler.codec.http.HttpResponseStatus.UNAUTHORIZED;
+
+import io.confluent.ksql.parser.tree.CreateConnector;
+import io.confluent.ksql.rest.entity.KsqlEntity;
+import io.confluent.ksql.services.ConnectClient.ConnectResponse;
+import io.confluent.ksql.statement.ConfiguredStatement;
+import java.util.Optional;
+import org.apache.kafka.connect.runtime.rest.entities.ConnectorInfo;
+
+/**
+ * An interface that allows to plug-in custom error handling for Connect server errors, such as 403
+ * Forbidden or 401 Unauthorized.
+ */
+public interface ConnectServerErrors {
+
+  /**
+   * Handles error response for a create connector request. This method dispatches to specific
```

Comment:
This new plugin should be applied to all connector requests, not just CREATE CONNECTOR.

Reply:
Done in https://github.com/confluentinc/ksql/pull/8480/commits/7ac1fe2c1b271aec0498c7f1315b34f0681bb7a4#diff-24e5d395473a57f7588579718e1f2209f549729a9d975c232ebedb3c0e52f905R81-R113.